.github/workflows/backend.yml

```diff
@@ -0,0 +1,445 @@
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      - name: Setup sentry env (python ${{ matrix.python-version }})
+        uses: ./.github/actions/setup-sentry
+        id: setup
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+          pip-cache-version: ${{ secrets.PIP_CACHE_VERSION }}
+          snuba: true
+          kafka: true
+
+      - name: Start symbolicator
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          echo $PWD
+          docker run \
+            -d \
+            -v $PWD/config/symbolicator/:/etc/symbolicator \
+            --network host \
+            --name symbolicator \
+            us.gcr.io/sentryio/symbolicator:nightly \
+            run -c /etc/symbolicator/config.yml
+          docker ps -a
+
+      - name: Run test
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make test-symbolicator
+
+      - name: Handle artifacts
+        uses: ./.github/actions/artifacts
+
+  typing:
+    needs: files-changed
+    name: backend typing
+    runs-on: ubuntu-20.04
+    timeout-minutes: 12
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: ./.github/actions/setup-python
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Since we don't call the setup-sentry action we need to install libxmlsec1-dev
+      - name: Setup backend typing
+        if: needs.changes.outputs.backend == 'true'
+        env:
+          SENTRY_LIGHT_BUILD: 1
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends libxmlsec1-dev
+          python setup.py install_egg_info
+          pip install wheel # GitHub Actions does not have `wheel` installed by default
+          pip install -U -e ".[dev]"
+
+      - name: Run backend typing (${{ steps.setup.outputs.matrix-instance-number }} of ${{ strategy.job-total }})
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make backend-typing
+
+  # This check runs once all dependant jobs have passed
+  # It symbolizes that all required Backend checks have succesfully passed
+  # This check is the only required Github check
+  backend-required-check:
+    needs:
+      [
+        api-docs,
+        backend,
+        cli,
+        lint,
+        migration,
+        plugins,
+        relay,
+        snuba,
+        symbolicator,
+        typing,
+      ]
```

Comment:
These are all the jobs defined above. All of them initially depend on `files-changed`.

Reply:
Can you confirm that if a single `needs` job fails, this job will also fail?

Another thing that would be nice to confirm is the behavior of 1) if we have 1 job skipped, rest success and 2) if all jobs are skipped

Reply:
Unless all of them pass, the required job will stay as skipped.

Over [here](https://github.com/getsentry/sentry/pull/32689#discussion_r832347061) shows that 1 check failed and it stayed skipped.

I had to re-run the failed check:
![image](https://user-images.githubusercontent.com/44410/162518448-90155f9c-2d31-4766-b9ae-281036b93e2e.png)

It even mentioned that "check to trigger VS" could get executed by re-running the single failure:
![image](https://user-images.githubusercontent.com/44410/162518521-b003ba58-5d70-42df-8242-00f4ae0e0417.png)

I can test your nice to confirm.